shared_model/validators/field_validator.hpp

```diff
@@ -48,27 +48,37 @@ namespace shared_model {
           ReasonsGroupType &reason,
           const interface::types::AccountIdType &account_id) const {
         if (not std::regex_match(account_id, account_id_)) {
```

Comment:
*musthave*

`account_id_` is bad name for regex. 

I understand that this is not part of this PR, but please, move all regexp out of constructor 
 (https://github.com/hyperledger/iroha/blob/67a82efcb776283f85a6b912f89d86cbba0a9837/shared_model/validators/field_validator.hpp#L39), make them as static members and give them meaningful names:
```
account_id_regexp
...
```

Reply:
Isn't it a bit unnecessary to add 'regexp' to a variable name, since its type is already std::regex?

Reply:
@nickaleks it is obvious for you in IDE, but when others read the code, it is quite hard to understand just by looking at the variable.

Reply:
Okay, also I think since this class does not have any state, we can as well use free functions in a proper namespace, as already done for example in address validation. But it is debatable because we tend to use classes everywhere.

Reply:
Also, this change would require its own pull request because right now if we declare regexes static, we have to define them after class declaration. So I will fix it when we separate declaration and definition of FieldValidator.

Reply:
@Warchant `std::regex` has a limited set of operations, so that if the variable is referenced in the code with `regex_match` for example, it is clear for the reader that referenced variable is indeed a regex.